### PR TITLE
feat: summary includes NonDet, etc.

### PR DESCRIPTION
**Pull request by JonathanStarup**

### CODE REVIEW

**Comment by magnus-madsen** [comment]
@mlutze You can discuss and merge when satisfied.

**Comment by magnus-madsen** [comment]
Note that the new base effects are currently used very very sparingly. See BaseEffects.

File: main/src/ca/uwaterloo/flix/language/ast/Purity.scala [hunk 1]
```diff
@@ -114,19 +114,32 @@ object Purity {
...
   def fromType(eff: Type)(implicit universe: Set[Symbol.EffectSym]): Purity = {
     evaluateFormula(eff) match {
       case set if set.isEmpty => Purity.Pure
-      case set if set.sizeIs == 1 && set.contains(Symbol.IO) => Purity.Impure
+      case set if set.forall(isJavaEffect) => Purity.Impure
       case _ => Purity.ControlImpure
     }
   }
 
+  def isJavaEffect(sym: Symbol.EffectSym): Boolean = sym match {
```
Comment (mlutze) [inline main/src/ca/uwaterloo/flix/language/ast/Purity.scala:130]: Surely this should be somewhere else, fx right next to where these all are declared.

Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/ast/Purity.scala:130]: @sockmaster27 Is adding this function also. You should coordinate and add it in one place.

Comment (sockmaster27) [reply main/src/ca/uwaterloo/flix/language/ast/Purity.scala:130]: I propose adding it to `Symbol`:  #8699

**Pull request closed without merging**